proposals/init.md

```diff
@@ -0,0 +1,582 @@
+    public init string LastName;
+}
+
+var s = new Student()
+{
+    FirstName = "Jarde",
+    LastName = "Parsons",
+}
+
+s.FirstName = "Jared"; // Error FirstName is readonly
+```
+
+In metadata these fields would be marked in the same way as `readonly` fields 
+but with an additional attribute or modreq to indicate they are `init` style
+fields. 
+
+**Resolution**
+LDM agrees this proposal is sound but overal the scenario felt disjoint from 
+properties. The decision was to proceed only with `init` properties for now. 
+This has a suitable level of flexibility as an `init` property can mutate a 
+`readonly` field on the declaring type of the property. This will be
+reconsidered if there is significant customer feedback that justifies the 
+scenario.
+
+### Allow init as a type modifier
+In the same way the `readonly` modifier can be applied to a `struct` to 
+automatically declare all fields as `readonly`, the `init` only modifier can
+be declared on a `struct` or `class` to automatically mark all fields as `init`.
+This means the following two type declarations are equivalent:
+
+```cs
+struct Point
+{
+    public init int X;
+    public init int Y;
+}
+
+// vs. 
+
+init struct Point
+{
+    public int X;
+    public int Y;
+}
+```
+
+**Resolution** 
+This feature is too *cute* here and conflicts with the `readonly struct` 
+feature on which it is based. The `readonly struct` feature is simple in that 
+it applies `readonly` to all members: fields, methods, etc ... The 
+`init struct` feature would only apply to properties. This actually ends up making
+it confusing for users. 
+
+Given that `init` is only valid on certain aspects of a type we rejected the 
+idea of having it as a type modifier.
+
+## Considerations
+
+### Compatibility
+The `init` feature is designed to be compatible with existin `get` only 
+properties. Specifically it is meant to be a completely addiditive change for 
+a property which is `get` only today but desires more flexbile object creation
+semantics.
+
+For example consider the following type:
+
+```cs
+class Name
+{
+    public string First { get; }
+    public string Last { get; }
+
+    public Name(string first, string last)
+    {
+        First = first;
+        Last = last;
+    }
+}
+```
+
+Adding `init` to these properties is a non-breaking change:
+
+```cs
+class Name
+{
+    public string First { get; init; }
+    public string Last { get; init; }
+
+    public Name(string first, string last)
+    {
+        First = first;
+        Last = last;
+    }
+}
+```
+
+### IL verification
+When .NET Core decides to re-implement IL verification the rules will need to be 
+adjusted to account for `init` members. This will need to be included in the 
+rule changes for non-mutating acess to `readonly` data.
+
+The IL verification rules will need to be broken into two parts: 
+
+1. Allowing `init` members to set a `readonly` field.
+1. Determining when an `init` member can be legally called.
```

Comment:
`2.`?

Reply:
I use `1.` for all of my markdown lists. The actual markdown rendering engine will replace them with the correctly numbered lists. Given that having them all labeled as `1.` makes re-ordering elements in the document easier because I don't have to re-number after re-arranging.

Reply:
I see! I already guessed that this was intentional when I saw it again a bit further down. Just makes it a bit odd when reading the source markdown 😄